segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/hdfs/HDFSSegmentHandle.java

```diff
@@ -48,7 +47,7 @@
     private HDFSSegmentHandle(String segmentName, boolean readOnly, List<FileDescriptor> files) {
         this.segmentName = Exceptions.checkNotNullOrEmpty(segmentName, "segmentName");
         this.readOnly = readOnly;
-        this.files = Exceptions.checkNotNullOrEmpty(files, "files");
```

Comment:
We should get rid of `FileDescriptor`. If we only have one file, then make this a `String` (and put back the null check).

Reply:
Fixed.